.github/workflows/build.yml

```diff
@@ -13,99 +13,115 @@ env:
-          wget https://github.com/pret/pokeheartgold/raw/workflows/assets/NitroSDK-3_2-060901.7z
-          unzip mwccarm.zip
-          mv mwccarm $GITHUB_WORKSPACE/tools
-          7z x NitroSDK-3_2-060901.7z
-          mv NitroSDK-3_2-060901/tools/bin $GITHUB_WORKSPACE/tools
-          mv NitroSDK-3_2-060901/include/nitro/specfiles/ARM7-TS.lcf.template $GITHUB_WORKSPACE/sub/
-          mv NitroSDK-3_2-060901/include/nitro/specfiles/ARM9-TS.lcf.template $GITHUB_WORKSPACE/
-          mv NitroSDK-3_2-060901/include/nitro/specfiles/mwldarm.response.template $GITHUB_WORKSPACE/
-        working-directory: ~
-
-      - name: Build HeartGold
+      - name: Build ROM
         env:
           GAME_VERSION: HEARTGOLD
           GAME_LANGUAGE: ENGLISH
           GAME_REVISION: 0
-        run: make -j${nproc}
+          NONMATCHING: 0
+          COMPARE: 1
+          BUILD_OUT: build/heartgold.us/
+        uses: ./.github/workflows/build-rom
+
+  build-soulsilver:
+    runs-on: ubuntu-22.04
 
-      - name: Build SoulSilver
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+
+      - name: Build ROM
         env:
           GAME_VERSION: SOULSILVER
           GAME_LANGUAGE: ENGLISH
           GAME_REVISION: 0
-        run: make -j${nproc}
+          NONMATCHING: 0
+          COMPARE: 1
+          BUILD_OUT: build/soulsilver.us/
+        uses: ./.github/workflows/build-rom
+
+  build-nonmatching:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+
+      - name: Build ROM
+        env:
+          GAME_VERSION: HEARTGOLD
+          GAME_LANGUAGE: ENGLISH
+          GAME_REVISION: 0
+          NONMATCHING: 1
+          COMPARE: 0
+        uses: ./.github/workflows/build-rom
+
+  post-merge:
+    runs-on: ubuntu-22.04
+    if: |
+      always() &&
+      (github.event_name == 'push') &&
+      contains(join(needs.*.result, ','), 'success')
+    needs: [build-heartgold, build-soulsilver, build-nonmatching]
+
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+
+      - name: Download HeartGold build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: HEARTGOLD-build
+
+      - name: Download SoulSilver build/ artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: SOULSILVER-build
+
+      - name: Extract build artifacts
+        run: |
+          mkdir HEARTGOLD_build/
+          mkdir SOULSILVER_build/
+          mkdir -p build/heartgold.us/
+          mkdir -p build/soulsilver.us/
+          mkdir -p sub/build/
+          tar -xzvhf HEARTGOLD_build.tar.gz -C HEARTGOLD_build/
+          tar -xzvhf SOULSILVER_build.tar.gz -C SOULSILVER_build/
+          cp -r HEARTGOLD_build/build/heartgold.us/ build/
+          cp -r SOULSILVER_build/build/soulsilver.us/ build/
+          cp -r HEARTGOLD_build/sub/build/ sub/
 
       - name: Webhook
-        if: ${{ github.event_name == 'push' }}
```

Comment:
not needed cause this requires push anyway, right??
did you test whether this works locally?

Reply:
This whole job is guarded now by `(github.event_name == 'push')`.